app.rb

```diff
@@ -1,123 +1,101 @@
+      create_rental
+    when '6'
+      list_rentals
+    when '7'
+      save
+      true
     else
-      create_teacher
+      puts 'That is not a valid option'
     end
-    puts "\n"
-  end
-
-  def create_student
-    print 'Age:'
-    age = gets.chomp
-    print 'Name:'
-    name = gets.chomp
-    print 'Has Parent Permission?[Y/N] '
-    permission = gets.chomp
-
-    @people.push(Student.new(age, name, parent_permission: permission))
-
-    puts 'Student Successfully Add!'
-    puts "\n"
   end
-
-  def create_teacher
-    print 'Age:'
-    age = gets.chomp
-    print 'Name: '
-    name = gets.chomp
-    print 'Specialization:'
-    specialization = gets.chomp
-
-    @people.push(Teacher.new(specialization, age, name, parent_permission: false))
-    puts 'Successfuly Add Teacher!'
-    puts "\n"
-  end
-
-  def create_rental
-    puts 'Select a book from the following list by number:'
-    book_list
-    book_index = gets.chomp.to_i
-
-    puts 'Select a person from the following list by number:'
-    people_list
-    people_index = gets.chomp.to_i
-
+  # rubocop:enable Metrics/CyclomaticComplexity
+
+  private
+
+  def rental_input
+    puts 'Select a book from the following list by number'
+    @book_storage.each_with_index { |book, i| puts "#{i}) #{book}" }
+    book_i = gets.chomp.to_i
+    puts
+    puts 'Select a person from the following list by number (not ID)'
+    @person_storage.each_with_index { |person, i| puts "#{i}) #{person}" }
+    person_i = gets.chomp.to_i
+    puts
     print 'Date: '
     date = gets.chomp
+    [date, person_i, book_i]
+  end
 
-    @rental.push(Rental.new(date, books[book_index - 1], people[people_index - 1]))
+  def add_rental(date, person_i, book_i)
+    rental = Rental.new(date, @person_storage.get_person_at_index(person_i), @book_storage.get_book_at_index(book_i))
+    @rentals.push(rental)
+  end
 
-    puts 'Rental create successfully!'
-    puts "\n"
+  def create_rental()
+    results = rental_input
+    add_rental(results[0], results[1], results[2])
+    puts 'Rental created successfullyy'
   end
 
-  def rental_list_by_id
+  def list_rentals
     print 'ID of person: '
-    id = gets.chomp.to_i
-    person1 = @people.select { |p| p.id == id }[0]
-    if person1
-      person1.rentals.each do |rental|
-        puts "Rental Date: #{rental.date}, Book: #{rental.book.title} by #{rental.person.name}"
-      end
-    else
-      puts 'Please enter correct ID!'
-    end
-    puts "\n"
+    id = gets.chomp
+
+    puts 'Rentals:'
+    @rentals.each { |rental| puts rental if rental.person.id == id.to_i }
   end
 
-  def exit
-    abort 'Thank you for using this App!'
+  def save
+    FileUtils.mkdir_p './data'
+    @person_storage.save
+    @book_storage.save
+    # save rentals
+    data = @rentals.map do |rental|
+      {
+        date: rental.date,
+        book_idx: @book_storage.books.index(rental.book),
+        person_idx: @person_storage.people.index(rental.person)
+      }
+    end
```

Comment:
Hi @shaqayq 
Great implementation of the different project requirements

- [ ] Please your app throws an error when a user exit the app  
### **Take a Look**
![exit](https://user-images.githubusercontent.com/55596970/194233750-06f27861-5fb7-480c-8106-fc5fbe2f768b.png)